lib/Backend/Vault/Kv/Internal.hs

```diff
@@ -252,29 +252,30 @@ type VaultTokenHeader = Header' '[Strict, Required] "X-Vault-Token" VaultToken
 -- $setup
 -- >>> import Network.HTTP.Client.TLS (tlsManagerSettings)
 -- >>> import Network.HTTP.Client     (newManager)
--- >>> import Servant.Clieng          (BaseUrl, Https)
+-- >>> import Servant.Client          (Scheme(Https), mkClientEnv)
+-- >>> import Servant.Client.Core     (BaseUrl(..))
+-- >>> import Control.Lens
 --
 -- >>> let vaultToken   = "TOKEN"
 -- >>> let vaultAddress = "vault.example.org"
 --
 -- >>>     manager     <- newManager tlsManagerSettings
--- >>> let clientEnv   <- mkClientEnv manager (BaseUrl Https vaultAddress 8200 "")
-
+-- >>> let clientEnv   =  mkClientEnv manager (BaseUrl Https vaultAddress 8200 "")
 
+-- @TODO - rewrite these doc tests (at this moment they don't type check)
 data Routes route =
   Routes
   { -- | To read a secret under a path use `readSecret`
-    -- >>> (routes clientEnv ^. readSecret) vaultToken "kv" ["testbed", "a1"]
+    --  (routes clientEnv ^. readSecret) vaultToken "kv" ["testbed", "a1"]
     _readSecret :: route
     :- "v1"
     :> Capture "mount" T.Text
     :> "data"
     :> VaultTokenHeader
     :> CaptureAll "segments" T.Text
-    :> QueryParam "version" Int
```

Comment:
We shouldn't remove `version` from `Routes._readSecret`. These routes are supposed to match Vault's HTTP API, so it's OK to have a `version` query parameter here.

In my other comment, I meant we should remove versions from our `BackendEffect` abstraction only.

Reply:
Then I don't understand how to remove `version` in `ReadSecret`. We have this code
```Haskell
...
ReadSecret path ->
        embedCatchMaybe path (readSecret env path) >>= \case
...
readSecret env = (I.routes env ^. I.readSecret) mount token
...
```
And we need the `version` parameter in `readSecret`.

Reply:
`readSecret`'s version parameter is optional, we can pass it `Nothing`:

```hs
ReadSecret path ->
        embedCatchMaybe path (readSecret env path Nothing) >>= \case
```

Reply:
Done